webapp/channels/src/packages/mattermost-redux/src/selectors/entities/channel_bookmarks.ts

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import type {ChannelBookmarksState} from '@mattermost/types/channel_bookmarks';
+import type {GlobalState} from '@mattermost/types/store';
+
+const EMPTY_BOOKMARKS = {};
+
+export const getChannelBookmarks = (state: GlobalState, channelId: string): ChannelBookmarksState['byChannelId'][string] => {
+    const bookmarks = state.entities.channelBookmarks.byChannelId[channelId];
+
+    if (!bookmarks) {
+        return EMPTY_BOOKMARKS;
+    }
+
+    return bookmarks;
+};
+
+export const getChannelBookmark = (state: GlobalState, channelId: string, bookmarkId: string) => {
+    return getChannelBookmarks(state, channelId)?.[bookmarkId];
```

Comment:
I fail to see why the `?` is needed as if no bookmarks the selector seems to be returning an empty object `EMPTY_BOOKMARKS`

Reply:
Good point. Was mainly from habit, but not indeed needed here.